src/event_logger.cpp

```diff
@@ -60,17 +60,90 @@ void EventLogger::createAsyncPelWithInventoryCallout(
     const std::optional<std::string> i_symFru,
     const std::optional<std::string> i_procedure)
 {
-    // TODO, implementation needs to be added.
-    (void)i_errorType;
-    (void)i_severity;
-    (void)i_callouts;
-    (void)i_fileName;
-    (void)i_funcName;
-    (void)i_internalRc;
-    (void)i_userData1;
-    (void)i_userData2;
     (void)i_symFru;
     (void)i_procedure;
+    try
+    {
+        if (i_callouts.empty())
+        {
+            logging::logMessage(
+                "Callout information is missing to create PEL.");
+            return;
+        }
+
+        if (m_errorMsgMap.find(i_errorType) == m_errorMsgMap.end())
+        {
+            logging::logMessage(
+                "PEL error type not found in the error message map.");
+            return;
+        }
+
+        std::string l_message = m_errorMsgMap.at(i_errorType);
+
+        std::string l_severity =
+            (m_severityMap.find(i_severity) != m_severityMap.end()
+                 ? m_severityMap.at(i_severity)
+                 : m_severityMap.at(types::SeverityType::Error));
+
+        std::string l_description("Description field missing in user data");
+
+        if (i_userData1 && ((*i_userData1).first.compare("DESCRIPTION") ==
+                            constants::STR_CMP_SUCCESS))
+        {
+            l_description = (*i_userData1).second;
+        }
+
+        sd_bus* l_sdBus = nullptr;
+        sd_bus_default(&l_sdBus);
+
+        types::InventoryCalloutData l_invCallout = i_callouts[0];
+        types::CalloutPriority l_priorityEnum = get<1>(l_invCallout);
+
+        std::string l_priority =
```

Comment:
Unless not required, please use const reference?

Reply:
Done